extension/src/ConnectionManager.ts

```diff
@@ -0,0 +1,92 @@
+import * as vscode from 'vscode';
+import { WebSocket } from "ws";
+import {ScClient} from 'ts-sc-client-ws';
+
+export enum HealthcheckStatus {
+    OK = 0,
+    FAIL
+}
+
+export class ConnectionManager {
+    client: ScClient;
+    status: HealthcheckStatus;
+    protected _statusBarItem: vscode.StatusBarItem;
+
+    constructor() {
+        this.client = undefined;
+        this.status = undefined;
+        this._statusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 100);
+        this._statusBarItem.command = "scs.connect";
+    }
+
+    get statusBarItem() { return this._statusBarItem; }
+
+    async connect(url: string) {
+        await this.disconnect();
+        await this.test(url);
+        while(this.status == undefined);
```

Comment:
This is kinda weird solution, but i've spend a couple of ~~hours~~ days to find a better way and... failed. Maybe you make this prettier (i.e. without (potentially) infinite loop)